etl/dags/extract_weather_data.py

```
@@ -0,0 +1,94 @@
+from datetime import timedelta, datetime
+from airflow import DAG
+from airflow.operators.python_operator import PythonOperator
+from airflow.utils.dates import days_ago
+import json
+from google.cloud.storage import Client
+import requests
+from deepmerge import always_merger
+import os
+
+BASE_URL = "http://api.weatherapi.com/v1/history.json?key={}&q={}+united+states&dt={}"
+# Get a weatherapi.com api key
+API_KEY = os.environ.get("API_WEATHER_KEY", "a70a4e2736644cdcb9d85348202404")
+BUCKET_NAME = os.environ.get("BUCKET_NAME", "default")
```

Should fail if env not set, to force correct configuration, instead of silently choosing an invalid bucket name.